src/chttpd/src/chttpd_db.erl

```diff
@@ -476,19 +477,59 @@ db_req(#httpd{method='POST', path_parts=[_, <<"_bulk_get">>]}=Req, Db) ->
             #doc_query_args{
                 options = Options
             } = bulk_get_parse_doc_query(Req),
-
-            {ok, Resp} = start_json_response(Req, 200),
-            send_chunk(Resp, <<"{\"results\": [">>),
-
-            lists:foldl(fun(Doc, Sep) ->
-                {DocId, Results, Options1} = bulk_get_open_doc_revs(Db, Doc,
-                                                                    Options),
-                bulk_get_send_docs_json(Resp, DocId, Results, Options1, Sep),
-                <<",">>
-            end, <<"">>, Docs),
-
-            send_chunk(Resp, <<"]}">>),
-            end_json_response(Resp)
+            % Decide whether to use multipart response or application/json
+            AcceptMixedMp = MochiReq:accepts_content_type("multipart/mixed"),
+            AcceptRelatedMp = MochiReq:accepts_content_type("multipart/related"),
+            AcceptMp = AcceptMixedMp orelse AcceptRelatedMp,
+            case AcceptMp of
+               false ->
+                  {ok, Resp} = start_json_response(Req, 200),
+                  send_chunk(Resp, <<"{\"results\": [">>),
+                  lists:foldl(fun(Doc, Sep) ->
+                      {DocId, Results, Options1} = bulk_get_open_doc_revs(Db, Doc,
+                                                                          Options),
+                      bulk_get_send_docs_json(Resp, DocId, Results, Options1, Sep),
+                      <<",">>
+                  end, <<"">>, Docs),
+                  send_chunk(Resp, <<"]}">>),
+                  end_json_response(Resp);
+               true ->
+                  OuterBoundary = bulk_get_multipart_boundary(),
+                  MpType = case AcceptMixedMp of
+                      true ->
+                          "multipart/mixed";
+                      _ ->
+                          "multipart/related"
+                  end,
+                  CType = {"Content-Type", MpType ++ "; boundary=\"" ++
+                                        ?b2l(OuterBoundary) ++  "\""},
+                  {ok, Resp} = start_chunked_response(Req, 200, [CType]),
+                  lists:foldl(fun(Doc, _Pre) ->
+                      case bulk_get_open_doc_revs(Db, Doc, Options) of
+                          {_, {ok, []}, _Options1} ->
+                              ok;
+                          {_, {ok, Results}, Options1} ->
+                              send_docs_multipart(bulk_get, Req, Results, Options1,
+                                                  OuterBoundary, Resp);
+                          {DocId, {error, {RevId, Error, Reason}}, _Options1} ->
+                              Json = ?JSON_ENCODE({[{<<"id">>, DocId},
+                                                    {<<"rev">>, RevId},
```

Comment:
Keep `[` on `?JSON_ENCODE` line but indent all the {..., ...} tuples 4 spaces to the right of `Json`

Like:
```
Json = ?JSON_ENCODE({[
    {<<"id">>, DocId},
    {<<"rev">>, RevId},
    ...
]}),
```